src/librustc_resolve/diagnostics.rs

```diff
@@ -93,6 +93,41 @@ mod bar {
 ```
 "##,
 
+E0254: r##"
+This error indicates that an imported symbol resolved to an already imported
+symbol.
+
+An example of this error:
+
+```
+// crate foo
+pub use foo::bar;
+pub mod foo; // note that there's a `pub` here
+```
+
+```
+// other program
+extern crate foo;
+
+use foo::*; // error, `foo::*` tries to bind `foo::foo` to `foo`
```

Comment:
I would discourage reusing the same name `foo` for such totally different roles, unless it is strictly necessary for replicating this particular error diagnostic.  Its okay for the example program to be a little bit longer, if it makes the example *clearer*.  (Also, if I read this correctly, you are using the `mod foo;` form, but have not included the text of the associated `foo.rs` file that would hold the body of that module?  That also seems overly difficult for the reader to understand; why not just put the definition of `mod foo` inline?)